src/thread/thread_d.hpp

```diff
@@ -390,7 +392,14 @@ namespace fc {
               /* NB: At least on Win64, this only catches a yield while in the body of 
                * a catch block; it fails to catch a yield while unwinding the stack, which 
                * is probably just as likely to cause crashes */
-              assert(std::current_exception() == std::exception_ptr());
+              if( std::current_exception() != std::exception_ptr() )
+              {
+                 std::stringstream stacktrace;
+                 print_stacktrace( stacktrace );
+                 elog( "Thread ${name} yielded in exception handler!\n${trace}",
+                       ("name",thread::current().name())("trace",stacktrace.str()) );
```

Comment:
I think it's best if we can log `std::current_exception()` as well, maybe `e.what()`? If it's a `fc::exception`, we can even get `e.to_detail_string()`.

By the way, I think it doesn't harm to print to `cerr` first, just in case when failing to write it to a log file due to E.G. disk space issues.

Reply:
`std::current_exception()` is a pointer to an unspecified type. There is no useful generic way to convert it to a string. Please remember that this piece of code is only meant to give us information about a situation that should never occur in the first place.

This is also the wrong place to care about disk space. Node operators can configure their nodes to log to console *and* logfile if they think it might be an issue.

Reply:
> wrong place to care about disk space

I agree, but I still think it's better to also print the stacktrace to console, since there are other reasons that the `elog()` call can fail, E.G. OOM so fail to copy the stacktrace to a string in the first place. ;-)

Reply:
Oh, come on. An OOM can occur just as easily in `print_stacktrace(std::cerr)`. We don't care about OOMs anywhere else we call `elog()`.

We *could* change `elog()` to generally log to console, but I think that would be a bad idea. That's what we have a configurable logging system for.